### ISSUE TITLE
build error: type defaults to ‘int’ in declaration of ‘amount’ [-Werror=implicit-int]

### ISSUE THREAD

**Issue by mikoim** [openzfs/zfs#10664]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Fedora
Distribution Version	| 32
Linux Kernel	| 5.7.11-250.vanilla.knurd.1.fc32.x86_64
Architecture	| x86_64
ZFS Version	| f1de1600d132666d03b3b73a2ab62695a0f60ead (build failed)
SPL Version	| f1de1600d132666d03b3b73a2ab62695a0f60ead (build failed)

### Describe the problem you're observing
I've got the following build error since 3442c2a02d.

### Describe how to reproduce the problem
```bash
cd
curl -s https://repos.fedorapeople.org/repos/thl/kernel-vanilla.repo | sudo tee /etc/yum.repos.d/kernel-vanilla.repo
sudo dnf --enablerepo=kernel-vanilla-stable update
git clone https://github.com/openzfs/zfs.git
cd zfs
./autogen.sh
./configure
make
```

### Include any warning/errors/backtraces from the system logs

- https://gist.github.com/mikoim/d385b12861b284e6afbf0c35118f454f
```
make -C /usr/src/kernels/5.7.11-250.vanilla.knurd.1.fc32.x86_64 M=`pwd`  CONFIG_ZFS=m modules
make[3]: Entering directory '/usr/src/kernels/5.7.11-250.vanilla.knurd.1.fc32.x86_64'
  CC [M]  /home/foobar/src/zfs/module/zfs/arc.o
In file included from ./include/trace/syscall.h:5,
                 from ./include/linux/syscalls.h:85,
                 from /home/foobar/src/zfs/include/os/linux/spl/sys/vnode.h:29,
                 from /home/foobar/src/zfs/include/os/linux/spl/sys/sunddi.h:32,
                 from /home/foobar/src/zfs/include/sys/zfs_context.h:61,
                 from /home/foobar/src/zfs/include/sys/spa.h:37,
                 from /home/foobar/src/zfs/module/zfs/arc.c:280:
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:388:10: warning: data definition has no type or storage class
  388 |  TP_ARGS(amount, arc_evict_count, aew_count),
      |          ^~~~~~
./include/linux/tracepoint.h:136:26: note: in definition of macro ‘TP_ARGS’
  136 | #define TP_ARGS(args...) args
      |                          ^~~~
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:388:10: error: type defaults to ‘int’ in declaration of ‘amount’ [-Werror=implicit-int]
  388 |  TP_ARGS(amount, arc_evict_count, aew_count),
      |          ^~~~~~
./include/linux/tracepoint.h:136:26: note: in definition of macro ‘TP_ARGS’
  136 | #define TP_ARGS(args...) args
      |                          ^~~~
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:388:18: error: type defaults to ‘int’ in declaration of ‘arc_evict_count’ [-Werror=implicit-int]
  388 |  TP_ARGS(amount, arc_evict_count, aew_count),
      |                  ^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:136:26: note: in definition of macro ‘TP_ARGS’
  136 | #define TP_ARGS(args...) args
      |                          ^~~~
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:388:35: error: type defaults to ‘int’ in declaration of ‘aew_count’ [-Werror=implicit-int]
  388 |  TP_ARGS(amount, arc_evict_count, aew_count),
      |                                   ^~~~~~~~~
./include/linux/tracepoint.h:136:26: note: in definition of macro ‘TP_ARGS’
  136 | #define TP_ARGS(args...) args
      |                          ^~~~
In file included from /home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:398,
                 from /home/foobar/src/zfs/include/os/linux/zfs/sys/trace_zfs.h:40,
                 from /home/foobar/src/zfs/module/zfs/arc.c:301:
./include/trace/define_trace.h:135: error: unterminated argument list invoking macro "DEFINE_EVENT"
  135 | #endif /* CREATE_TRACE_POINTS */
      | 
In file included from /home/foobar/src/zfs/include/os/linux/zfs/sys/trace_zfs.h:40,
                 from /home/foobar/src/zfs/module/zfs/arc.c:301:
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:386:1: error: type defaults to ‘int’ in declaration of ‘DEFINE_EVENT’ [-Werror=implicit-int]
  386 | DEFINE_EVENT(zfs_arc_wait_for_eviction_class, name, \
      | ^~~~~~~~~~~~
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:390:1: note: in expansion of macro ‘DEFINE_ARC_WAIT_FOR_EVICTION_EVENT’
  390 | DEFINE_ARC_WAIT_FOR_EVICTION_EVENT(zfs_arc__wait__for__eviction);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/foobar/src/zfs/module/zfs/arc.c:330:17: error: conflicting types for ‘arc_evict_count’
  330 | static uint64_t arc_evict_count;
      |                 ^~~~~~~~~~~~~~~
In file included from ./include/trace/syscall.h:5,
                 from ./include/linux/syscalls.h:85,
                 from /home/foobar/src/zfs/include/os/linux/spl/sys/vnode.h:29,
                 from /home/foobar/src/zfs/include/os/linux/spl/sys/sunddi.h:32,
                 from /home/foobar/src/zfs/include/sys/zfs_context.h:61,
                 from /home/foobar/src/zfs/include/sys/spa.h:37,
                 from /home/foobar/src/zfs/module/zfs/arc.c:280:
/home/foobar/src/zfs/include/os/linux/zfs/sys/trace_arc.h:388:18: note: previous declaration of ‘arc_evict_count’ was here
  388 |  TP_ARGS(amount, arc_evict_count, aew_count),
      |                  ^~~~~~~~~~~~~~~
./include/linux/tracepoint.h:136:26: note: in definition of macro ‘TP_ARGS’
  136 | #define TP_ARGS(args...) args
      |                          ^~~~
In file included from /home/foobar/src/zfs/include/sys/zfs_context.h:64,
                 from /home/foobar/src/zfs/include/sys/spa.h:37,
                 from /home/foobar/src/zfs/module/zfs/arc.c:280:
/home/foobar/src/zfs/module/zfs/arc.c: In function ‘arc_wait_for_eviction’:
/home/foobar/src/zfs/include/os/linux/spl/sys/trace.h:75:2: error: implicit declaration of function ‘trace_zfs_arc__wait__for__eviction’ [-Werror=implicit-function-declaration]
   75 |  trace_zfs_##name((arg1), (arg2), (arg3))
      |  ^~~~~~~~~~
/home/foobar/src/zfs/module/zfs/arc.c:5091:4: note: in expansion of macro ‘DTRACE_PROBE3’
 5091 |    DTRACE_PROBE3(arc__wait__for__eviction,
      |    ^~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:267: /home/foobar/src/zfs/module/zfs/arc.o] Error 1
make[4]: *** [scripts/Makefile.build:488: /home/foobar/src/zfs/module/zfs] Error 2
make[3]: *** [Makefile:1732: /home/foobar/src/zfs/module] Error 2
make[3]: Leaving directory '/usr/src/kernels/5.7.11-250.vanilla.knurd.1.fc32.x86_64'
make[2]: *** [Makefile:48: modules-Linux] Error 2
make[2]: Leaving directory '/home/foobar/src/zfs/module'
make[1]: *** [Makefile:870: all-recursive] Error 1
make[1]: Leaving directory '/home/foobar/src/zfs'
make: *** [Makefile:733: all] Error 2
```

```
$ gcc --version
gcc (GCC) 10.2.1 20200723 (Red Hat 10.2.1-1)
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by qjim-github** [comment]
Problem is with wrong definition of 'zfs_arc_wait_for_eviction_class' in include/os/linux/zfs/sys/trace_arc.h file.
Unfortunately I'm not able to past text with tabs, so all formatting is lost. Therefore 'patch -l' or manual editing is necessary.
Simply: TP_PROTO line must end with added '\' and newline and TP_ARGS line end must be changed from '),' to '))'
##############################################
`--- a/include/os/linux/zfs/sys/trace_arc.h  2020-08-03 14:57:25.334563884 +0200`
`+++ b/include/os/linux/zfs/sys/trace_arc.h  2020-08-03 14:57:31.341230473 +0200`
`@@ -382,10 +382,11 @@ DECLARE_EVENT_CLASS(zfs_arc_wait_for_evi`
`/* END CSTYLED */`
` `
`/* BEGIN CSTYLED */`
`-#define        DEFINE_ARC_WAIT_FOR_EVICTION_EVENT(name) \`
`-DEFINE_EVENT(zfs_arc_wait_for_eviction_class, name, \`
`-       TP_PROTO(uint64_t amount, uint64_t arc_evict_count, uint64_t aew_count),`
`-       TP_ARGS(amount, arc_evict_count, aew_count),`
`+#define        DEFINE_ARC_WAIT_FOR_EVICTION_EVENT(name)                \`
`+DEFINE_EVENT(zfs_arc_wait_for_eviction_class, name,            \`
`+       TP_PROTO(uint64_t amount, uint64_t arc_evict_count,     \`
`+                uint64_t aew_count),                           \`
`+       TP_ARGS(amount, arc_evict_count, aew_count))`
`/* END CSTYLED */`
`DEFINE_ARC_WAIT_FOR_EVICTION_EVENT(zfs_arc__wait__for__eviction);`
` `

**Comment by mikoim** [comment]
fixed 309d20882fbc3c3a870a6c7c58ba95503c12b254
